test/suites/secondary_http.bash

```diff
@@ -0,0 +1,87 @@
+start_http_server() {
+    local host="127.0.0.1"
+    local port="8080"
+    local url="http://${host}:${port}/"
+    export CCACHE_SECONDARY_STORAGE="${url}"
+
+    mkdir "secondary"
+    echo "=== TEST ${CURRENT_TEST} ===" >>http-server.log
+    "${HTTP_SERVER}" --bind "${host}" --directory "secondary" "${port}" >>http-server.log 2>&1 &
+
+    echo "=== TEST ${CURRENT_TEST} ===" >>http-client.log
+    "${HTTP_CLIENT}" "${url}" >>http-client.log 2>&1 || test_failed_internal "Cannot connect to server"
+}
+
+SUITE_secondary_http_PROBE() {
+    if ! "${HTTP_SERVER}" --help >/dev/null 2>&1; then
+        echo "cannot execute ${HTTP_SERVER} - Python 3 might be missing"
+    fi
+}
+
+SUITE_secondary_http_SETUP() {
+    unset CCACHE_NODIRECT
+
+    start_http_server
+
+    generate_code 1 test.c
+}
+
+SUITE_secondary_http_TEARDOWN() {
+    terminate_all_children
+}
+
```

Comment:
I guess this is superfluous now that `test/run` runs `terminate_all_children`? Would you like to keep it for symmetry with `start_http_server`?

Reply:
I was under the assumption that each `TEST` invocation also cleans the server working dir. But that's not the case. I'll simplify the logic.

Reply:
I completely removed `TEARDOWN` and by using subdirs, still have the complete state archived on failure